bson/bsonrw/json_scanner_test.go

```diff
@@ -280,28 +300,28 @@ func TestJsonScannerValidInputs(t *testing.T) {
 
 			for _, token := range tc.tokens {
 				c, err := js.nextToken()
+				expectNoError(t, err, tc.desc)
```

Comment:
This changes the order of the error checking in these `jsonScanner` tests so that the "no error" assertions happen before any value assertions. Previously, if an error was thrown, it was possibly not printed or appeared after the associated value assertion, which are both confusing when debugging tests.